README.md

```diff
@@ -54,3 +54,18 @@ The JavaScript interface currently only supports IQL-strict queries.
 Make sure [babashka](https://github.com/babashka/babashka) is installed. Then 
 run the tests via `bb test`. Dialect-specific tests can be run with 
 `bb test:clj` and `bb test:cljs`. 
+
+#### Dependency upgrades
+
+We use `clj-nix` to support tamper-proof reproducible builds of the dependencies for all environments.
+When you upgrade a package in `deps.edn`, it is necessary to update `deps-lock.json` as well, so that
+the nix build universe has knowledge of the hash fingerprints of the new deps version tarballs.
```

Comment:
In general, yes for automation. Post-commit hooks I particularly dislike because they interrupt any workflow with assurances about every workflow (imagine if you do `git commit` and suddenly your system is hanging waiting for Nix to build for the first time). I would rather have check in github actions that prevents merge unless you do the manual action when your PR is ready; or better yet, find a way to do a dependabot-style automatic PR opening thing.

Reply:
+1 for CI automation. I'm experimenting with [Jujutsu](https://github.com/martinvonz/jj) over git, and I don't think it supports hooks yet.